Sources/ProcedureKit/Group.swift

```diff
@@ -52,6 +52,16 @@ open class GroupProcedure: Procedure, ProcedureQueueDelegate {
         get { return groupStateLock.withCriticalScope { _groupChildren } }
     }
 
+    /// Override of errors
+    public override var errors: [Error] {
+        get { return groupStateLock.withCriticalScope { _groupErrors.fatal } }
+        set {
+            groupStateLock.withCriticalScope {
+                _groupErrors.fatal = newValue
+            }
+        }
+    }
+
```

Comment:
I think essentially, that what is happening here is that `GroupProcedure` sort of handles errors a little bit differently so that it can have the concept of error recovery. This means, disambiguating between fatal and recovered errors. However, the error mechanisms in `Procedure` which are activated by `cancel(withError:)` and not overridden by `GroupProcedure` therefore mean that these errors are not picked up.

Although, given this explanation, I don't see why this has only just started happening now - possibly a false positive which the warnings in Xcode 8.3 has helped shine a light on.

So, possible solution would be to make the internal `_procedureDidCancel()` _micro-optimization_ in `Procedure` accept the additional errors argument. This would allow `GroupProcedure` to append the error to it's fatal array immediately.